azure-devops/azext_devops/test/repos/test_ref.py

```diff
@@ -0,0 +1,74 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import unittest
+
+try:
+    # Attempt to load mock (works on Python 3.3 and above)
+    from unittest.mock import patch
+except ImportError:
+    # Attempt to load mock (works on Python version below 3.3)
+    from mock import patch
+
+from azext_devops.dev.common.services import clear_connection_cache
+from azext_devops.dev.repos.ref import (list_refs, create_ref, delete_ref, update_ref)
+
+
+class TestRefMethods(unittest.TestCase):
+
+    _TEST_DEVOPS_ORGANIZATION = 'https://AzureDevOpsCliTest.visualstudio.com'
+
+    def setUp(self):
+        self.list_refs_patcher = patch('vsts.git.v4_0.git_client.GitClient.get_refs')
+        self.update_ref_patcher = patch('vsts.git.v4_0.git_client.GitClient.update_refs')
+
+        # start the patchers
+        self.mock_list_refs = self.list_refs_patcher.start()
+        self.mock_update_ref = self.update_ref_patcher.start()
+
+        # clear connection cache before running each test
+        clear_connection_cache()
+
+    def tearDown(self):
+        self.mock_list_refs.stop()
+        self.mock_update_ref.stop()
+
+    def test_list_refs(self):
+        response = list_refs(devops_organization=self._TEST_DEVOPS_ORGANIZATION,
+                             project='sample project',
+                             detect='off')
+        # assert
+        self.mock_list_refs.assert_called_once()
```

Comment:
We are trying to move to more strict tests. See #318. 

In your case the commands in itself don't have any code logic just call the sdk. You can use **assert_called_once_with()**  to make more strict asserts. 

Anything else that you think would help or any suggestions in this area are welcome.